File: backend/src/main/java/com/backend/domain/Item/Item/entity/Item.java
package com.backend.domain.Item.Item.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;
    private String content;

    private int price;
    private int stock;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist(){
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate(){
        this.updatedAt= LocalDateTime.now();
    }
}

File: frontend/src/type/item.ts
//추후 백엔드와 연동 시 Dto로 변경해주면 된다. ex)ItemDto
//지금은 임시 데이터

export type Item = {
    id: number;
    name: string;
    content: string;
    price: number;
    imageUrl: string;
    createdAt:string;
    updatedAt:string;
  };
  